internal/database/sqlcommon/pin_sql.go

```diff
@@ -141,24 +141,25 @@ func (s *SQLCommon) GetPins(ctx context.Context, filter database.Filter) (messag
 
 }
 
-func (s *SQLCommon) SetPinDispatched(ctx context.Context, sequence int64) (err error) {
+func (s *SQLCommon) UpdatePins(ctx context.Context, filter database.Filter, update database.Update) (err error) {
 
 	ctx, tx, autoCommit, err := s.beginOrUseTx(ctx)
 	if err != nil {
 		return err
 	}
 	defer s.rollbackTx(ctx, tx, autoCommit)
 
-	_, err = s.updateTx(ctx, tx, sq.
-		Update("pins").
-		Set("dispatched", true).
-		Where(sq.Eq{
-			sequenceColumn: sequence,
-		}),
-		func() {
-			s.callbacks.OrderedCollectionEvent(database.CollectionPins, fftypes.ChangeEventTypeUpdated, sequence)
-		},
-	)
+	query, err := s.buildUpdate(sq.Update("pins"), update, pinFilterFieldMap)
+	if err != nil {
+		return err
+	}
+
+	query, err = s.filterUpdate(ctx, "", query, filter, pinFilterFieldMap)
+	if err != nil {
+		return err
+	}
+
+	_, err = s.updateTx(ctx, tx, query, nil /* no change events filter based update */)
```

Comment:
So this type does reliably emit ChangeEventTypeCreated. It theoretically emits ChangeEventTypeDeleted, although I don't know of anywhere we call DeletePin. It now never emits ChangeEventTypeUpdated though.

I wonder if having change events at all is useful, or if it's confusing to emit them inconsistently.

Reply:
This is common across the codebase for these update-multiple events.

We don't have a need for them where we use them currently, and it's hard in SQL without detailed DB-specific coding to find out what rows have been updated. There's also a performance overhead